tests/playwright/src/model/pages/kubernetes-context-page.ts

```diff
@@ -0,0 +1,112 @@
+  readonly heading: Locator;
+  readonly content: Locator;
+  readonly contextTable: Locator;
+
+  constructor(page: Page) {
+    super(page, 'Kubernetes');
+    this.heading = this.page.getByLabel('Title', { exact: true });
+    this.content = this.page.getByLabel('Content');
+    this.contextTable = this.content.getByLabel('Contexts');
+  }
+
+  async pageIsEmpty(): Promise<boolean> {
+    const emptyHeading = this.page.getByRole('heading', { name: 'No Kubernetes contexts found', exact: true });
+    return (await emptyHeading.count()) > 0;
+  }
+
+  async getContextRowByName(name: string): Promise<Locator> {
+    return this.contextTable.getByLabel(name, { exact: true });
+  }
+
+  async isContextDefault(name: string): Promise<boolean> {
+    const row = await this.getContextRowByName(name);
+    if (row === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const bannerText = await row.getByLabel('Current Context').textContent();
+
+    return bannerText === 'Current Context';
+  }
+
+  async isContextReachable(name: string): Promise<boolean> {
+    const row = await this.getContextRowByName(name);
+    if (row === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const contextReachable = row.getByLabel('Context Reachable');
+
+    return (await contextReachable.count()) > 0;
+  }
+
+  async setDefaultContext(name: string): Promise<void> {
+    const contextRow = await this.getContextRowByName(name);
+    if (contextRow === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const switchButton = contextRow.getByLabel('Set as Current Context');
+    await playExpect(switchButton).toBeEnabled();
+    await switchButton.click();
+  }
+
+  async deleteContext(name: string, handleConfirmation: boolean = true): Promise<void> {
+    const contextRow = await this.getContextRowByName(name);
+    if (contextRow === undefined) {
+      throw Error(`Context: '${name}' does not exist`);
+    }
+    const deleteButton = contextRow.getByLabel('Delete Context');
+    await playExpect(deleteButton).toBeEnabled();
+    await deleteButton.click();
+    if (handleConfirmation) {
+      await handleConfirmationDialog(this.page, 'Delete Context');
+    }
+  }
+
+  async getContextName(row: Locator): Promise<Locator> {
+    return row.getByLabel('Context Name', { exact: true });
+  }
+
+  async getContextCluster(row: Locator): Promise<Locator> {
+    return row.getByLabel('Context Cluster');
+  }
+
+  async getContextServer(row: Locator): Promise<Locator> {
+    return row.getByLabel('Context Server');
+  }
+
+  async getContextUser(row: Locator): Promise<Locator> {
+    return row.getByLabel('Context User');
+  }
+
+  async getContextNamespace(row: Locator): Promise<Locator> {
+    return row.getByLabel('Context Namespace');
+  }
+
+  async getSetCurrentContextButton(row: Locator): Promise<Locator> {
+    return row.getByLabel('Set as Current Context');
+  }
+}
```

Comment:
There's no reason these methods should need to take a locator param nor for that matter should they return a locator param, I understand that it's easier to do this in order to `await playExpect(locator).toHaveText(text)` but the naming of the methods suggests they return the actual value of that field, in this case it should return a string type.

Also it would be even better if you only provided the actual name/context as a param, not a locator.

Additionally given the current implementation why are they even returning a Promise<Locator>? there is nothing async in these, which makes it all the more confusing.

Reply:
I tried to address this, but am not sure if in the correct way. I wasn`t able to find a way to remove the Promises from the returns of these functions as the async construction requires it.